src/main.ipynb

```diff
@@ -1213,20 +1213,21 @@
    ],
    "source": [
     "n = bin_list(10)\n",
-    "d = [0,0,0,0,0,0,0,0,0,0]\n",
+    "d = [0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0]\n",
     "total = 0\n",
     "i = 1\n",
     "for vector in n\n",
     "    k = missrow(vector)\n",
+    "    t = misstotal(vector)\n",
     "    if k != 0\n",
-    "        d[k] += 1\n",
+    "        d[k] += 0.1^(t)*0.9^(10-t)\n",
     "    end\n",
     "end\n",
     "for element in d\n",
     "    total += i*element\n",
     "    i+=1\n",
     "end\n",
-    "final = total/1024"
+    "total"
```

Comment:
What we want here is to apply `missrow` to every vector in `n`, and calculate the mean of the values. This can be achieved by
```julia
values = map(missrow, n)
total = mean(values)
```
`map(func, iterable)` takes a function and an iterable (like an array or vector) and applies the `func` function to every element in the iterable, then returns a new iterable with the new values. `mean()` then calculate the mean of all values in an iterable.

The less code there are, the fewer places to make mistakes 😉

Reply:
Edit: I forgot that each sequence is not equally likely to occur. This requires small modification:
```julia
n = bin_list(10)
p = 0.9
values = map(vec -> missrow(vec) * p^(10 - misstotal(vec)) * (1 - p)^misstotal(vec), n)
total = sum(values)
```
We multiply the `missrow` value of every vector with its probability, and then take the **sum** instead of mean. The result is now the same as before.